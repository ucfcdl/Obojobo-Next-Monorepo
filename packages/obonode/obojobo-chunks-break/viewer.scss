@import '~styles/includes';

.obojobo-draft--chunks--break {
	text-align: center;

	hr {
		border: none;
		border-bottom: 1px solid $color-shadow;
		margin-bottom: 1em;
		margin-top: 1em;
		width: 30%;
	}
}
