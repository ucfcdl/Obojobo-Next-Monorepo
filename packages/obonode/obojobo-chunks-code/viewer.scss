@import '../obojobo-document-engine/src/scss';

.obojobo-draft--chunks--code {
	pre {
		font-size: 1em;
		padding-left: 2em;
		padding-right: 2em;
	}

	.obo-text {
		display: block;
	}
}
