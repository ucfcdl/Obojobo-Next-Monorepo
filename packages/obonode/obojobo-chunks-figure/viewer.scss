@import '~styles/includes';

.obojobo-draft--chunks--figure {
	text-align: center;

	figure {
		margin: 0;
		padding: 0;
		border: 1px solid red;
	}

	.container {
		display: inline-block;
		height: auto;
		position: relative;
		// width: 18em;

		img {
			background: $color-bg;
			border: none;
			display: block;
			width: 100%;
		}

		.img-placeholder {
			border: 1px solid $color-shadow;
		}
	}

	figcaption {
		color: $color-text-minor;
		font-size: 0.9em;
	}

	&.small {
		figure {
			height: 100%;
			margin: 0;
			padding: 0;
			width: 100%;
		}

		.container {
			width: 18em;
		}
	}

	&.medium {
		figure {
			height: 100%;
			margin: 0;
			padding: 0;
			width: 100%;
		}

		.container {
			width: $dimension-column-width;
		}
	}

	&.large {
		figure {
			padding-top: 0;

			img {
				display: block;
			}

			.img-placeholder {
				border-left: none;
				border-right: none;
			}
		}

		.container {
			width: 100%;
		}
	}
}
