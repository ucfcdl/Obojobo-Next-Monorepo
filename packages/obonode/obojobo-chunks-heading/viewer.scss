@import '~styles/includes';

// .obojobo-draft--chunks--heading
// {
// }
