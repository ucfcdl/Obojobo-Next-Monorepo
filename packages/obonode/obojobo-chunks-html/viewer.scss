@import '~styles/includes';

.obojobo-draft--chunks--html {
	display: block;
	margin: 1em auto;
}
