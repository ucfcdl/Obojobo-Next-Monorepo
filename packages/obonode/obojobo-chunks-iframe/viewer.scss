@import '~styles/includes';

.obojobo-draft--chunks--iframe {
	display: block;
	font-size: 0.8em;
	margin: 0 auto;
	max-width: $dimension-editor;
	text-align: left;
	width: 100%;

	iframe {
		height: 20em;
		margin: 0 auto;
		width: 100%;
	}
}
