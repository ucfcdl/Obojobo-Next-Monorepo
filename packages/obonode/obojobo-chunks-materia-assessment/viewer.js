import adapter from 'obojobo-chunks-materia/adapter'
import Common from 'obojobo-document-engine/src/scripts/common'
import ViewerComponent from './viewer-component'

Common.Registry.registerModel('ObojoboDraft.Chunks.MateriaAssessment', {
	adapter: adapter,
	componentClass: ViewerComponent,
	type: 'chunk'
})
