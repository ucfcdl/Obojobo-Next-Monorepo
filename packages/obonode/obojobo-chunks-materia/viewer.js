import adapter from './adapter'
import Common from 'obojobo-document-engine/src/scripts/common'
import ViewerComponent from './viewer-component'

Common.Registry.registerModel('ObojoboDraft.Chunks.Materia', {
	adapter: adapter,
	componentClass: ViewerComponent,
	type: 'chunk'
})
