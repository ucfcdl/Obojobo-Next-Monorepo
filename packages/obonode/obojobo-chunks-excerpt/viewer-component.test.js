import React from 'react'
import renderer from 'react-test-renderer'

import Excerpt from './viewer-component'
import OboModel from 'obojobo-document-engine/src/scripts/common/models/obo-model'

const chunkJSON = {
	id: 'id',
	type: 'ObojoboDraft.Chunks.Excerpt',
	content: {
		bodyStyle: 'filled-box',
		width: 'medium',
		font: 'sans',
		lineHeight: 'moderate',
		fontSize: 'smaller',
		topEdge: 'normal',
		bottomEdge: 'normal',
		effect: false
	},
	children: [
		{
			id: 'id-1',
			type: 'ObojoboDraft.Chunks.Excerpt',
			subtype: 'ObojoboDraft.Chunks.Excerpt.ExcerptContent',
			content: {
				bodyStyle: 'filled-box',
				width: 'medium',
				font: 'sans',
				lineHeight: 'moderate',
				fontSize: 'smaller',
				topEdge: 'normal',
				bottomEdge: 'normal',
				effect: false
			},
			children: [
				{
					type: 'ObojoboDraft.Chunks.Text',
					content: {},
					children: [
						{
							type: 'ObojoboDraft.Chunks.Text',
							subtype: 'ObojoboDraft.Chunks.Text.TextLine',
							content: {
								indent: 0
							},
							children: [
								{
									text: ''
								}
							]
						}
					]
				}
			]
		},
		{
			id: 'id-2',
			type: 'ObojoboDraft.Chunks.Excerpt',
			subtype: 'ObojoboDraft.Chunks.Excerpt.CitationText',
			content: {
				indent: 0,
				hangingIndent: 0
			},
			children: [
				{
					id: 'id-3',
					type: 'ObojoboDraft.Chunks.Excerpt',
					subtype: 'ObojoboDraft.Chunks.Excerpt.CitationLine',
					content: {
						indent: 0,
						hangingIndent: 0,
						align: 'center'
					},
					children: [
						{
							text: ''
						}
					]
				}
			]
		}
	]
}

require('./viewer') // used to register this oboModel

describe('Excerpt', () => {
	test('Excerpt component', () => {
		const model = OboModel.create(chunkJSON)
		const moduleData = {
			focusState: {}
		}
		const component = renderer.create(<Excerpt model={model} moduleData={moduleData} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('Excerpt component with effect', () => {
		const moduleData = {
			focusState: {}
		}
		const model = OboModel.create({
			...chunkJSON,
			content: {
				...chunkJSON.content,
				effect: true
			}
		})
		const component = renderer.create(<Excerpt model={model} moduleData={moduleData} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('Excerpt component with citation', () => {
		const moduleData = {
			focusState: {}
		}
		const model = OboModel.create({
			...chunkJSON,
			content: {
				...chunkJSON.content,
				citation: 'Placeholder text'
			}
		})

		const component = renderer.create(<Excerpt model={model} moduleData={moduleData} />)

		const cite = component.root.findByType('cite')
		expect(cite.children[0]).toBe('Placeholder text')

		const tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('Excerpt component with no citation', () => {
		const moduleData = {
			focusState: {}
		}
		const model = OboModel.create({
			...chunkJSON,
			content: {
				...chunkJSON.content,
				citation: ''
			}
		})

		const component = renderer.create(<Excerpt model={model} moduleData={moduleData} />)

		const cite = component.root.findAllByType('cite')
		expect(cite.length).toBe(0)

		const tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
