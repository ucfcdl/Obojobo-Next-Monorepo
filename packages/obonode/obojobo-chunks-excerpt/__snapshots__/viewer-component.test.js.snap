// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Excerpt Excerpt component 1`] = `
<div
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.Excerpt"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-normal is-bottom-edge-type-normal is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
  >
    <blockquote
      cite="test"
    >
      <div
        className="excerpt-content"
      >
        <div
          className="wrapper"
        >
          <div
            className="component"
            data-focussed={false}
            data-id="id-1"
            data-obo-component={true}
            data-type="ObojoboDraft.Chunks.Excerpt"
            id="obo-id-1"
            tabIndex={-1}
          >
            <div
              className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-normal is-bottom-edge-type-normal is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
            >
              <blockquote
                cite="test"
              >
                <div
                  className="excerpt-content"
                >
                  <div
                    className="wrapper"
                  />
                  <div
                    className="overlay"
                  />
                </div>
              </blockquote>
            </div>
          </div>
          <div
            className="component"
            data-focussed={false}
            data-id="id-2"
            data-obo-component={true}
            data-type="ObojoboDraft.Chunks.Excerpt"
            id="obo-id-2"
            tabIndex={-1}
          >
            <div
              className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-undefined is-bottom-edge-type-undefined is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
            >
              <blockquote
                cite="test"
              >
                <div
                  className="excerpt-content"
                >
                  <div
                    className="wrapper"
                  >
                    <div
                      className="component"
                      data-focussed={false}
                      data-id="id-3"
                      data-obo-component={true}
                      data-type="ObojoboDraft.Chunks.Excerpt"
                      id="obo-id-3"
                      tabIndex={-1}
                    >
                      <div
                        className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-undefined is-bottom-edge-type-undefined is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
                      >
                        <blockquote
                          cite="test"
                        >
                          <div
                            className="excerpt-content"
                          >
                            <div
                              className="wrapper"
                            />
                            <div
                              className="overlay"
                            />
                          </div>
                        </blockquote>
                      </div>
                    </div>
                  </div>
                  <div
                    className="overlay"
                  />
                </div>
              </blockquote>
            </div>
          </div>
        </div>
        <div
          className="overlay"
        />
      </div>
    </blockquote>
  </div>
</div>
`;

exports[`Excerpt Excerpt component with citation 1`] = `
<div
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.Excerpt"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-normal is-bottom-edge-type-normal is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
  >
    <blockquote
      cite="test"
    >
      <div
        className="excerpt-content"
      >
        <div
          className="wrapper"
        >
          <div
            className="component"
            data-focussed={false}
            data-id="id-1"
            data-obo-component={true}
            data-type="ObojoboDraft.Chunks.Excerpt"
            id="obo-id-1"
            tabIndex={-1}
          >
            <div
              className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-normal is-bottom-edge-type-normal is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
            >
              <blockquote
                cite="test"
              >
                <div
                  className="excerpt-content"
                >
                  <div
                    className="wrapper"
                  />
                  <div
                    className="overlay"
                  />
                </div>
              </blockquote>
            </div>
          </div>
          <div
            className="component"
            data-focussed={false}
            data-id="id-2"
            data-obo-component={true}
            data-type="ObojoboDraft.Chunks.Excerpt"
            id="obo-id-2"
            tabIndex={-1}
          >
            <div
              className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-undefined is-bottom-edge-type-undefined is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
            >
              <blockquote
                cite="test"
              >
                <div
                  className="excerpt-content"
                >
                  <div
                    className="wrapper"
                  >
                    <div
                      className="component"
                      data-focussed={false}
                      data-id="id-3"
                      data-obo-component={true}
                      data-type="ObojoboDraft.Chunks.Excerpt"
                      id="obo-id-3"
                      tabIndex={-1}
                    >
                      <div
                        className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-undefined is-bottom-edge-type-undefined is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
                      >
                        <blockquote
                          cite="test"
                        >
                          <div
                            className="excerpt-content"
                          >
                            <div
                              className="wrapper"
                            />
                            <div
                              className="overlay"
                            />
                          </div>
                        </blockquote>
                      </div>
                    </div>
                  </div>
                  <div
                    className="overlay"
                  />
                </div>
              </blockquote>
            </div>
          </div>
        </div>
        <div
          className="overlay"
        />
      </div>
      <cite>
        Placeholder text
      </cite>
    </blockquote>
  </div>
</div>
`;

exports[`Excerpt Excerpt component with effect 1`] = `
<div
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.Excerpt"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-normal is-bottom-edge-type-normal is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-showing-effect"
  >
    <blockquote
      cite="test"
    >
      <div
        className="excerpt-content"
      >
        <div
          className="wrapper"
        >
          <div
            className="component"
            data-focussed={false}
            data-id="id-1"
            data-obo-component={true}
            data-type="ObojoboDraft.Chunks.Excerpt"
            id="obo-id-1"
            tabIndex={-1}
          >
            <div
              className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-normal is-bottom-edge-type-normal is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
            >
              <blockquote
                cite="test"
              >
                <div
                  className="excerpt-content"
                >
                  <div
                    className="wrapper"
                  />
                  <div
                    className="overlay"
                  />
                </div>
              </blockquote>
            </div>
          </div>
          <div
            className="component"
            data-focussed={false}
            data-id="id-2"
            data-obo-component={true}
            data-type="ObojoboDraft.Chunks.Excerpt"
            id="obo-id-2"
            tabIndex={-1}
          >
            <div
              className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-undefined is-bottom-edge-type-undefined is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
            >
              <blockquote
                cite="test"
              >
                <div
                  className="excerpt-content"
                >
                  <div
                    className="wrapper"
                  >
                    <div
                      className="component"
                      data-focussed={false}
                      data-id="id-3"
                      data-obo-component={true}
                      data-type="ObojoboDraft.Chunks.Excerpt"
                      id="obo-id-3"
                      tabIndex={-1}
                    >
                      <div
                        className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-undefined is-bottom-edge-type-undefined is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
                      >
                        <blockquote
                          cite="test"
                        >
                          <div
                            className="excerpt-content"
                          >
                            <div
                              className="wrapper"
                            />
                            <div
                              className="overlay"
                            />
                          </div>
                        </blockquote>
                      </div>
                    </div>
                  </div>
                  <div
                    className="overlay"
                  />
                </div>
              </blockquote>
            </div>
          </div>
        </div>
        <div
          className="overlay"
        />
      </div>
    </blockquote>
  </div>
</div>
`;

exports[`Excerpt Excerpt component with no citation 1`] = `
<div
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.Excerpt"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-normal is-bottom-edge-type-normal is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
  >
    <blockquote
      cite="test"
    >
      <div
        className="excerpt-content"
      >
        <div
          className="wrapper"
        >
          <div
            className="component"
            data-focussed={false}
            data-id="id-1"
            data-obo-component={true}
            data-type="ObojoboDraft.Chunks.Excerpt"
            id="obo-id-1"
            tabIndex={-1}
          >
            <div
              className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-normal is-bottom-edge-type-normal is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
            >
              <blockquote
                cite="test"
              >
                <div
                  className="excerpt-content"
                >
                  <div
                    className="wrapper"
                  />
                  <div
                    className="overlay"
                  />
                </div>
              </blockquote>
            </div>
          </div>
          <div
            className="component"
            data-focussed={false}
            data-id="id-2"
            data-obo-component={true}
            data-type="ObojoboDraft.Chunks.Excerpt"
            id="obo-id-2"
            tabIndex={-1}
          >
            <div
              className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-undefined is-bottom-edge-type-undefined is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
            >
              <blockquote
                cite="test"
              >
                <div
                  className="excerpt-content"
                >
                  <div
                    className="wrapper"
                  >
                    <div
                      className="component"
                      data-focussed={false}
                      data-id="id-3"
                      data-obo-component={true}
                      data-type="ObojoboDraft.Chunks.Excerpt"
                      id="obo-id-3"
                      tabIndex={-1}
                    >
                      <div
                        className="text-chunk obojobo-draft--chunks--excerpt pad is-body-style-type-filled-box is-top-edge-type-undefined is-bottom-edge-type-undefined is-width-medium is-font-sans is-font-style-undefined is-line-height-type-moderate is-font-size-smaller is-not-showing-effect"
                      >
                        <blockquote
                          cite="test"
                        >
                          <div
                            className="excerpt-content"
                          >
                            <div
                              className="wrapper"
                            />
                            <div
                              className="overlay"
                            />
                          </div>
                        </blockquote>
                      </div>
                    </div>
                  </div>
                  <div
                    className="overlay"
                  />
                </div>
              </blockquote>
            </div>
          </div>
        </div>
        <div
          className="overlay"
        />
      </div>
    </blockquote>
  </div>
</div>
`;
